ledger/complete/ledger.go

```diff
@@ -241,3 +246,62 @@ func (l *Ledger) Checkpointer() (*wal.Checkpointer, error) {
 	}
 	return checkpointer, nil
 }
+
+// ExportCheckpointAt exports a checkpoint at specific state commitment after applying migrations and returns the new state (after migration) and any errors
+func (l *Ledger) ExportCheckpointAt(state ledger.State, migrations []ledger.Migration, targetPathFinderVersion uint8, outputFilePath string) (ledger.State, error) {
+
+	// get trie
+	t, err := l.forest.GetTrie(ledger.RootHash(state))
+	if err != nil {
+		return nil, fmt.Errorf("cannot get try at the given state commitment: %w", err)
+	}
+
+	// get all payloads
+	payloads := t.AllPayloads()
+	payloadSize := len(payloads)
+
+	// migrate payloads
+	for i, migrate := range migrations {
+		payloads, err = migrate(payloads)
+		if err != nil {
+			return nil, fmt.Errorf("error applying migration: %w", err)
```

Comment:
I feel like migrations should have a name or some identifier?

Otherwise how do I know which migration failed?

Reply:
I added migration index for now (02e002f)